vc-got.el

```diff
@@ -413,20 +415,19 @@ files on disk."
         (table (list "HEAD")))
     (vc-got-with-worktree default-directory
       (with-temp-buffer
-        (when (zerop (vc-got--call "ref" "-l"))
-          (goto-char (point-min))
-          (while (re-search-forward re nil t)
-            (push (match-string 2) table))
-          table)))))
+        (vc-got-command t 0 nil "ref" "-l")
+        (goto-char (point-min))
+        (while (re-search-forward re nil t)
+          (push (match-string 2) table))
+        table))))
 
 (defun vc-got--branch (name)
   "Try to create and switch to the branch called NAME."
+  ;; TODO: does branch change files or does it require update
```

Comment:
IIRC this function is supposed to just create a new branch, not switch to an existing one, so it does not require an update.  I think it's expected to fail if the target branch already exists.

Reply:
Yeah, I was thinking does it need "  (let (process-file-side-effects)" at the start

Reply:
That's a good question.  Technically this changes some files in `.got` and in the bare repository, but I doubt this is relevant (noone has these files open in a emacs buffer, and it's a kind of implementation detail).  So yeah, it would make sense to me to add `(let (process-file-side-effects)` here.